BayeuxClient.agent.lib.nut

```diff
@@ -0,0 +1,936 @@
+        }
+    }
+
+    function _send(message, callback, connect = false) {
+        local headers = {
+            "Content-Type" : "application/json;charset=UTF-8"
+        }
+
+        foreach (k, v in _configuration.requestHeaders) {
+            headers[k] <- v;
+        }
+
+        foreach (name, cookie in _cookies) {
+            if (!("Cookie" in headers)) {
+                headers["Cookie"] <- "";
+            }
+            headers["Cookie"] += cookie + "; ";
+        }
+
+        local jsonMsg = null;
+        try {
+            jsonMsg = http.jsonencode(message);
+        } catch (e) {
+            _logError("Cannot convert message to JSON string: " + e);
+            return;
+        }
+
+        _log("Sending message: " + jsonMsg);
+
+        local httpRequest = http.post(_configuration.url, headers, jsonMsg);
+        if (connect) {
+            _connectHttpRequest = httpRequest;
+        } else {
+            _extraHttpRequest = [httpRequest, callback];
+        }
+
+        local sent = function(response) {
+            _onSent(response, callback, connect);
+        }.bindenv(this);
+
+        local timeout = _configuration.requestTimeout;
+        if (connect) {
+            // _advice.timeout is in msec, so we need to convert it to sec
+            timeout += _advice.timeout / 1000;
+        }
+
+        // TODO: Is it OK to set null as a streaming callback?
+        httpRequest.sendasync(sent, null, timeout);
+    }
+
+    function _onSent(response, callback, connect) {
+        if (connect) {
+            _connectHttpRequest = null;
+        } else {
+            _extraHttpRequest = null;
+        }
+
+        _log("Response received:");
+        _logTable(response);
+
+        if (!_statusIsOk(response.statuscode)) {
+            callback(Bayeux.Error(BAYEUX_CLIENT_ERROR_TYPE.TRANSPORT_FAILED, response.statuscode));
+            return;
+        }
+
+        // TODO: Handle the cookies properly
+        foreach (k, v in response.headers) {
+            if (k.tolower() == "set-cookie") {
+                try {
+                    local cookieName = split(v, "=")[0];
+                    _cookies[cookieName] <- v;
+                } catch(e) {
+                    _logError(e);
+                }
+            }
+        }
+
+        local tableMsg = null;
+        try {
+            tableMsg = http.jsondecode(response.body);
+        } catch (e) {
+            _logError("Response body is not a valid JSON: " + e);
+            _logError("Message: " + response.body);
+            callback(Bayeux.Error(BAYEUX_CLIENT_ERROR_TYPE.BAYEUX_ERROR, "Response body is not a valid JSON: " + e));
+            return;
+        }
+
+        if (tableMsg != null) {
+            callback(null);
+            _handleMessages(tableMsg);
+        }
+
+        if (!_isDisconnecting) {
+            _processQueue();
+        }
+    }
+
+    // Check HTTP status
+    function _statusIsOk(status) {
+        return status / 100 == 2;
```

Comment:
status.tointeger()?

Reply:
We always get an integer statuscode from http API, so I don't see any reason to make a conversion

Reply:
I see ok